### ISSUE TITLE
Parent counters with newtheoremrep

### ISSUE THREAD

**Issue by a3nm** [PierreSenellart/apxproof#7]
To make it easier to migrate documents to apxproof, it would be nice if `\newtheoremrep` could support parent counters, the way it is done in amsthm. E.g., I'd like to be able to write:

```
\newtheoremrep{theorem}{Theorem}[section]
\newtheoremrep{lemma}[theorem]{Lemma}
```

and have theorems numbered like "Theorem i.j", where i is the section number and j is the theorem number, and likewise for Lemmas (with the same numbering).

Right now this doesn't work, and using `\newtheoremrep{theorem}{Theorem}[section]` does not number theorems following sections. I can cheat by using `\newtheorem` for some other environment, e.g., `\newtheorem{definition}{Definition}[definition]`, and then use the definition counter in `\newtheoremrep`, but that's a bit ugly. Is it possible to fix this?

**Comment by PierreSenellart** [comment]
Implemented in 74f882212a5f8b913b892cbd97c110313affe94d. Can you check if the version of the develop branch meets your needs? If so, I will release a new version.

**Comment by a3nm** [comment]
I just tested it, and it seems to give the same result on my example
document (compared using pdftotext) with the new version as it did with
the workaround. So I think this is working fine! Thanks. :)

**Comment by PierreSenellart** [comment]
Thanks. Released 1.0.5.
